roles/haproxy/tasks/main.yml

```
@@ -1,5 +1,11 @@
 ---
 
+- name: Gather facts for hosts postres_cluster
+  ansible.builtin.setup:
+  delegate_to: "{{ item }}"
+  delegate_facts: true
+  loop: "{{ groups['postgres_cluster'] }}"
```

```suggestion
- name: Gather facts for hosts postres_cluster
  ansible.builtin.setup:
  delegate_to: "{{ item }}"
  delegate_facts: true
  loop: "{{ groups['postgres_cluster'] }}"
  when: hostvars[groups['postgres_cluster'][0]].ansible_hostname is not defined
```

perform the task only when there are no facts